src/test/java/SeleniumTestngTest.java

```diff
@@ -2,23 +2,135 @@
 
-        // Write your assertions in the after the steps of scenario are executed to validate results
-        Assert.assertTrue(true);
+        // Verify the presence of the text "A/B Test Control"
+        WebElement heading = browser.findElement(By.xpath("//h3[contains(text(),'A/B Test Control')]"));
+        Assert.assertTrue(heading.isDisplayed(), "The heading 'A/B Test Control' is not displayed!");
+    }
+
+    @Test
+    public void testAddRemoveElements() {
+        // Navigate to the "Add/Remove Elements" page
+        browser.get("https://the-internet.herokuapp.com/add_remove_elements/");
+
+        // Add three "Delete" buttons
+        WebElement addButton = browser.findElement(By.cssSelector("button[onclick='addElement()']"));
+        addButton.click();
+        addButton.click();
+        addButton.click();
+
+        // Check that three "Delete" buttons are added and visible
+        Assert.assertEquals(browser.findElements(By.cssSelector(".added-manually")).size(), 3, "Not all 'Delete' buttons are present!");
+
+        // Remove all "Delete" buttons
+        for (WebElement deleteButton : browser.findElements(By.cssSelector(".added-manually"))) {
+            deleteButton.click();
+        }
+
+        // Check that buttons are removed
+        Assert.assertTrue(browser.findElements(By.cssSelector(".added-manually")).isEmpty(), "Not all 'Delete' buttons were removed!");
+    }
+
+    @Test
+    public void testCheckboxes() {
+        // Navigate to the "Checkboxes" page
+        browser.get("https://the-internet.herokuapp.com/checkboxes");
+
+        // Select all checkboxes
+        for (WebElement checkbox : browser.findElements(By.cssSelector("input[type='checkbox']"))) {
+            if (!checkbox.isSelected()) {
+                checkbox.click();
+            }
+            Assert.assertTrue(checkbox.isSelected(), "Checkbox is not selected!");
+        }
     }
 
-    // Write the rest of TEST METHODS according to the task here, each method checking one scenario described in README.md file
-    // In the end you should have a set of test methods each of them describing some specific scenario
+    @Test
+    public void testDropdown() {
+        // Navigate to the "Dropdown" page
+        browser.get("https://the-internet.herokuapp.com/dropdown");
+
+        // Select "Option 2"
+        WebElement dropdown = browser.findElement(By.cssSelector("#dropdown"));
+        dropdown.click();
+        WebElement option2 = browser.findElement(By.xpath("//option[@value='2']"));
+        option2.click();
+
+        // Verify the option is selected
+        Assert.assertTrue(option2.isSelected(), "Option 2 is not selected!");
+    }
+
+    @Test
+    public void testFormAuthentication() {
+        // Navigate to the "Form Authentication" page
+        browser.get("https://the-internet.herokuapp.com/login");
 
+        // Enter login and password
+        browser.findElement(By.cssSelector("#username")).sendKeys("tomsmith");
```

Comment:
Hardcoded user/password values can be extracted at least into local variables = would be much easier to find them in case the credentials will change in the future